### PR TITLE
chore: Update Scala to 2.13.13

### PR DESCRIPTION
**Pull request by He-Pin**
Update Scala to 2.13.13

### CODE REVIEW

**Comment by Roiocam** [comment]
CI check shows some compilation errors.

```
[02-27 06:55:56.144] [error] (osgi / Test / compileIncremental) Compilation failed
```

File: project/PekkoDisciplinePlugin.scala [hunk 1]
```diff
@@ -74,16 +74,19 @@ object PekkoDisciplinePlugin extends AutoPlugin {
     "pekko-stream-tests-tck",
     "pekko-testkit")
 
-  val defaultScalaOptions = "-Wconf:cat=unused-nowarn:s,any:e"
+  lazy val defaultScalaOptions = Def.setting(CrossVersion.partialVersion(scalaVersion.value) match {
+    case Some((2, 12)) => "-Wconf:cat=unused-nowarn:s,any:e"
+    case _             => "-Wconf:cat=unused-nowarn:s,cat=lint-named-booleans:s,cat=other-shadowing:s,any:e"
```
Comment (raboof) [inline project/PekkoDisciplinePlugin.scala:79]: was it impossible to fix the warnings?

Comment (He-Pin) [reply project/PekkoDisciplinePlugin.scala:79]: Possible, but then need to change a lot of code.

Comment (pjfanning) [reply project/PekkoDisciplinePlugin.scala:79]: some of the code that doesn't compile appears to be generated - I think we may need to skip this upgrade and report bugs to Scala team

Comment (raboof) [reply project/PekkoDisciplinePlugin.scala:79]: gotcha - maybe create an issue for it here for starters and refer to that from here?

**Review by pjfanning — APPROVED** [review]
lgtm - the changes to the compile settings seem alright

**Review by raboof — APPROVED** [review]
the link validator failure looks unrelated and also present on master, filed #1158

**Review by raboof — COMMENTED** [review]
oh - turns out the link validator is actually correct for a change, and  https://github.com/apache/incubator-pekko/blob/main/scripts/link-validator.conf#L64 needs an update.

**Pull request merged**